x-pack/plugins/reporting/kibana.jsonc

```diff
@@ -24,17 +24,17 @@
       "embeddable",
       "screenshotMode",
       "share",
-      "features"
+      "features", 
+      "spaces"
```

Comment:
@elastic/kibana-security From a quick look, most (all?) plugins only use `spaces` as an `optional` dependency. Are we fine with this change moving the plugin to the required dependency list for the `reporting` plugin? 

(I know this may have changed multiple times with our serverless approach to spaces but I don't remember what the final outcome was)

Reply:
If it's not too expensive and feasible, I would still encourage consumers to treat Spaces as an optional dependency until we are completely certain about the future of Spaces. I'm not particularly thrilled about this recommendation myself, but it would make our lives much easier should we eventually decide to support an offering/deployment model without Spaces.

Reply:
So, @rshen91, what did you decide? Is making Spaces a required dependency a must-have for you for some reason, or can you still keep it optional as it was before?